sdk/python/kfp/dsl/component_factory.py

```diff
@@ -94,10 +107,17 @@ def make_index_url_options(pip_index_urls: Optional[List[str]]) -> str:
     index_url = pip_index_urls[0]
     extra_index_urls = pip_index_urls[1:]
 
-    options = [f'--index-url {index_url} --trusted-host {index_url}']
-    options.extend(
-        f'--extra-index-url {extra_index_url} --trusted-host {extra_index_url}'
-        for extra_index_url in extra_index_urls)
+    options = [f'--index-url {index_url}']
+    options.extend(f'--extra-index-url {extra_index_url}'
+                   for extra_index_url in extra_index_urls)
+
+    if pip_trusted_hosts is None or len(pip_trusted_hosts) == 0:
+        options.extend([f'--trusted-host {index_url}'])
+        options.extend(f'--trusted-host {extra_index_url}'
+                       for extra_index_url in extra_index_urls)
```

Comment:
So, by default we trust both `index_url` and each `extra_index_url` even though we don't explicitly state so via the `--trusted-host` :thinking:

I understand that this is here to keep the backward behavior, it's just interesting :slightly_smiling_face:

Reply:
Yes!

Reply:
```
  --trusted-host <hostname>   Mark this host or host:port pair as trusted, even though it does not have valid or any HTTPS.

```

I understand this is the original way of doing things, but I'm not a fan of this at all, the default behavior should be the "secure path", and users can opt in to provide "--trusted-host" if they want to.

Correct me if I'm wrong, but currently there is no way to *not* add any `--trusted-host` right? 

If so, I would at least like to see, when user provides `pip_trusted_hosts=[]` then we don't add `--trusted-host` at all, and rely on the user provisioning the appropriate certs for a proper tls validation. 

@diegolovison let me know if my understanding here is correct.

cc @chensun  thoughts?

Reply:
Done. Please read the description.